server/json-rpc/src/lib.rs

```diff
@@ -45,30 +43,18 @@ impl JsonRpcServerTrait for JsonRpcServer {
         // `subgraph_add` handler.
         let add_provider = provider.clone();
         let add_logger = logger.clone();
-        let subgraph_added = AtomicBool::new(false);
         let add_handler = move |params: SubgraphAddParams| {
             let provider = add_provider.clone();
             info!(add_logger, "Received subgraph_add request"; "params" => params.to_string());
             provider
-                .add(format!("/ipfs/{}", params.ipfs_hash))
+                .add(params.name, format!("/ipfs/{}", params.ipfs_hash))
                 .map_err(|e| json_rpc_error(0, e.to_string()))
                 .map(|_| Ok(Value::Null))
                 .flatten()
         };
         handler.add_method("subgraph_add", move |params: Params| {
             let handler = add_handler.clone();
```

Comment:
The clone is necessary because we can't move `add_handler` through the outer closure straight into the inner one. But I tried to clean up this code by moving the handler logic into a function.